app/controllers/application_controller.rb

```diff
@@ -1,2 +1,18 @@
 class ApplicationController < ActionController::Base
+	before_action :switch_locale
+	
+	def default_url_options
+      {locale: I18n.locale}
+  	end
+
+	private
+
+  	def switch_locale
+      I18n.locale = extract_locale || I18n.default_locale
+  	end
+
+  	def extract_locale
+      parsed_locale = params[:locale]
+    	I18n.available_locales.map(&:to_s).include?(parsed_locale) ? parsed_locale : nil
+  	end
```

Comment:
Notes: You have to have 2 spaces

```suggestion
  before_action :switch_locale
	
  def default_url_options
    {locale: I18n.locale}
  end

  private

  def switch_locale
    I18n.locale = extract_locale || I18n.default_locale
  end

  def extract_locale
    parsed_locale = params[:locale]
    I18n.available_locales.map(&:to_s).include?(parsed_locale) ? parsed_locale : nil
  end
```

Reply:
![image](https://user-images.githubusercontent.com/68467813/90381330-cbb8b580-e0af-11ea-98f3-bc2e8ae08161.png)

done